components/tag-list/tag-list.js

```diff
@@ -222,15 +267,55 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 		return items.filter(({ isHidden }) => !isHidden);
 	}
 
-	_getTagListItems() {
+	async _getTagListItems() {
 		const slot = this.shadowRoot && this.shadowRoot.querySelector('slot');
 		if (!slot) return;
-		return slot.assignedNodes({ flatten: true }).filter((node) => {
+
+		const results = await this._filterAsync(slot.assignedNodes({ flatten: true }), async node => {
 			if (node.nodeType !== Node.ELEMENT_NODE) return false;
+			await node.updateComplete;
```

Comment:
I was finding that it was necessary to await an updateComplete from the slot items, otherwise on clicking the "add tag" button in the demo, it wasn't getting the newest item and wasn't behaving properly (didn't have the role of `listitem` yet).